File: redome/src/main/java/br/org/cancer/modred/vo/PedidoExameDoadorNacionalVo.java
package br.org.cancer.modred.vo;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 
 * Classe utilizada para criação da consulta de doador - contato passivo.
 * 
 * @author evandro.gomes
 *
 */
public class PedidoExameDoadorNacionalVo implements Serializable {

	private static final long serialVersionUID = 4860435637249185865L;
	
	private Long idTarefa;
	private Long tipoTarefa;
	
	private Long idBusca;
	
	private Long rmr;
	private String nomePaciente;
	
	private Long idSolicitacao;
	private String tipoSolicitacaoDescricao;
	private String statusSolicitacaoDescricao;
	
	private Long dmr;
	private Long idDoador;
	private Long tipoDoador;
	private String nomeDoador;
	private String nomeLaboratorio;

	private Long idPedidoExame;
	private String tipoExameDescricao;
	
	private Long idPedidoEnriquecimento;
	private Long idPedidoContato;
	private Long idPedidoContatoSms;
	
	private String statusDescricao;
	private LocalDateTime dataCriacao;

	public PedidoExameDoadorNacionalVo() {}

	/**
	 * @return the idTarefa
	 */
	public Long getIdTarefa() {
		return idTarefa;
	}

	/**
	 * @param idTarefa the idTarefa to set
	 */
	public void setIdTarefa(Long idTarefa) {
		this.idTarefa = idTarefa;
	}

	/**
	 * @return the tipoTarefa
	 */
	public Long getTipoTarefa() {
		return tipoTarefa;
	}

	/**
	 * @param tipoTarefa the tipoTarefa to set
	 */
	public void setTipoTarefa(Long tipoTarefa) {
		this.tipoTarefa = tipoTarefa;
	}

	/**
	 * @return the idBusca
	 */
	public Long getIdBusca() {
		return idBusca;
	}

	/**
	 * @param idBusca the idBusca to set
	 */
	public void setIdBusca(Long idBusca) {
		this.idBusca = idBusca;
	}

	/**
	 * @return the rmr
	 */
	public Long getRmr() {
		return rmr;
	}

	/**
	 * @param rmr the rmr to set
	 */
	public void setRmr(Long rmr) {
		this.rmr = rmr;
	}

	/**
	 * @return the nomePaciente
	 */
	public String getNomePaciente() {
		return nomePaciente;
	}

	/**
	 * @param nomePaciente the nomePaciente to set
	 */
	public void setNomePaciente(String nomePaciente) {
		this.nomePaciente = nomePaciente;
	}

	/**
	 * @return the idSolicitacao
	 */
	public Long getIdSolicitacao() {
		return idSolicitacao;
	}

	/**
	 * @param idSolicitacao the idSolicitacao to set
	 */
	public void setIdSolicitacao(Long idSolicitacao) {
		this.idSolicitacao = idSolicitacao;
	}

	/**
	 * @return the idDoador
	 */
	public Long getIdDoador() {
		return idDoador;
	}

	/**
	 * @param idDoador the idDoador to set
	 */
	public void setIdDoador(Long idDoador) {
		this.idDoador = idDoador;
	}

	/**
	 * @return the tipoDoador
	 */
	public Long getTipoDoador() {
		return tipoDoador;
	}

	/**
	 * @param tipoDoador the tipoDoador to set
	 */
	public void setTipoDoador(Long tipoDoador) {
		this.tipoDoador = tipoDoador;
	}

	/**
	 * @return the dmr
	 */
	public Long getDmr() {
		return dmr;
	}

	/**
	 * @param dmr the dmr to set
	 */
	public void setDmr(Long dmr) {
		this.dmr = dmr;
	}

	/**
	 * @return the nomeDoador
	 */
	public String getNomeDoador() {
		return nomeDoador;
	}

	/**
	 * @param nomeDoador the nomeDoador to set
	 */
	public void setNomeDoador(String nomeDoador) {
		this.nomeDoador = nomeDoador;
	}

	/**
	 * @return the nomeLaboratorio
	 */
	public String getNomeLaboratorio() {
		return nomeLaboratorio;
	}

	/**
	 * @param nomeLaboratorio the nomeLaboratorio to set
	 */
	public void setNomeLaboratorio(String nomeLaboratorio) {
		this.nomeLaboratorio = nomeLaboratorio;
	}

	/**
	 * @return the idPedidoExame
	 */
	public Long getIdPedidoExame() {
		return idPedidoExame;
	}

	/**
	 * @param idPedidoExame the idPedidoExame to set
	 */
	public void setIdPedidoExame(Long idPedidoExame) {
		this.idPedidoExame = idPedidoExame;
	}

	/**
	 * @return the tipoExameDescricao
	 */
	public String getTipoExameDescricao() {
		return tipoExameDescricao;
	}

	/**
	 * @param tipoExameDescricao the tipoExameDescricao to set
	 */
	public void setTipoExameDescricao(String tipoExameDescricao) {
		this.tipoExameDescricao = tipoExameDescricao;
	}

	/**
	 * @return the statusDescricao
	 */
	public String getStatusDescricao() {
		return statusDescricao;
	}

	/**
	 * @param statusDescricao the statusDescricao to set
	 */
	public void setStatusDescricao(String statusDescricao) {
		this.statusDescricao = statusDescricao;
	}

	/**
	 * @return the idPedidoEnriquecimento
	 */
	public Long getIdPedidoEnriquecimento() {
		return idPedidoEnriquecimento;
	}

	/**
	 * @param idPedidoEnriquecimento the idPedidoEnriquecimento to set
	 */
	public void setIdPedidoEnriquecimento(Long idPedidoEnriquecimento) {
		this.idPedidoEnriquecimento = idPedidoEnriquecimento;
	}

	/**
	 * @return the idPedidoContato
	 */
	public Long getIdPedidoContato() {
		return idPedidoContato;
	}

	/**
	 * @param idPedidoContato the idPedidoContato to set
	 */
	public void setIdPedidoContato(Long idPedidoContato) {
		this.idPedidoContato = idPedidoContato;
	}

	/**
	 * @return the tipoSolicitacaoDescricao
	 */
	public String getTipoSolicitacaoDescricao() {
		return tipoSolicitacaoDescricao;
	}

	/**
	 * @param tipoSolicitacaoDescricao the tipoSolicitacaoDescricao to set
	 */
	public void setTipoSolicitacaoDescricao(String tipoSolicitacaoDescricao) {
		this.tipoSolicitacaoDescricao = tipoSolicitacaoDescricao;
	}

	/**
	 * @return the statusSolicitacaoDescricao
	 */
	public String getStatusSolicitacaoDescricao() {
		return statusSolicitacaoDescricao;
	}

	/**
	 * @param statusSolicitacaoDescricao the statusSolicitacaoDescricao to set
	 */
	public void setStatusSolicitacaoDescricao(String statusSolicitacaoDescricao) {
		this.statusSolicitacaoDescricao = statusSolicitacaoDescricao;
	}

	/**
	 * @return the dataCriacao
	 */
	public LocalDateTime getDataCriacao() {
		return dataCriacao;
	}

	/**
	 * @param dataCriacao the dataCriacao to set
	 */
	public void setDataCriacao(LocalDateTime dataCriacao) {
		this.dataCriacao = dataCriacao;
	}

	/**
	 * @return the idPedidoContatoSms
	 */
	public Long getIdPedidoContatoSms() {
		return idPedidoContatoSms;
	}

	/**
	 * @param idPedidoContatoSms the idPedidoContatoSms to set
	 */
	public void setIdPedidoContatoSms(Long idPedidoContatoSms) {
		this.idPedidoContatoSms = idPedidoContatoSms;
	}
	
	
}

File: redome-front/src/app/shared/vo/pedido.exame.doador.nacional.vo.ts
import { BaseEntidade } from '../base.entidade';
import { ConvertUtil } from '../util/convert.util';

export class PedidoExameDoadorNacionalVo extends BaseEntidade {
 
    private _idTarefa: number;
    private _tipoTarefa: number;
    private _idBusca: number;
    private _rmr: number;
    private _nomePaciente: string;
    private _idSolicitacao: number;
    private _tipoSolicitacaoDescricao: string;
    private _statusSolicitacaoDescricao: string;
    private _idDoador: number;
    private _tipoDoador: number;
    private _dmr: number;
    private _nomeDoador: string;
    private _nomeLaboratorio: string;
    private _idPedidoExame: number;
    private _dataCriacao: Date;
    private _tipoExameDescricao: string;
    private _statusDescricao: string;
    private _idPedidoEnriquecimento: number;
    private _idPedidoContato: number;
    private _idPedidoContatoSms: number;

    /**
     * Getter idTarefa
     * @return {number}
     */
    public get idTarefa(): number {
        return this._idTarefa;
    }

    /**
     * Getter tipoTarefa
     * @return {number}
     */
    public get tipoTarefa(): number {
        return this._tipoTarefa;
    }

    /**
     * Getter idBusca
     * @return {number}
     */
    public get idBusca(): number {
        return this._idBusca;
    }

    /**
     * Getter rmr
     * @return {number}
     */
    public get rmr(): number {
        return this._rmr;
    }

    /**
     * Getter nomePaciente
     * @return {string}
     */
    public get nomePaciente(): string {
        return this._nomePaciente;
    }

    /**
     * Getter idSolicitacao
     * @return {number}
     */
    public get idSolicitacao(): number {
        return this._idSolicitacao;
    }

    /**
     * Getter idDoador
     * @return {number}
     */
    public get idDoador(): number {
        return this._idDoador;
    }

    /**
     * Getter tipoDoador
     * @return {number}
     */
    public get tipoDoador(): number {
        return this._tipoDoador;
    }

    /**
     * Getter dmr
     * @return {number}
     */
    public get dmr(): number {
        return this._dmr;
    }

    /**
     * Getter nomeDoador
     * @return {string}
     */
    public get nomeDoador(): string {
        return this._nomeDoador;
    }

    /**
     * Getter nomeLaboratorio
     * @return {string}
     */
    public get nomeLaboratorio(): string {
        return this._nomeLaboratorio;
    }

    /**
     * Getter idPedidoExame
     * @return {number}
     */
    public get idPedidoExame(): number {
        return this._idPedidoExame;
    }

    /**
     * Getter tipoExameDescricao
     * @return {string}
     */
    public get tipoExameDescricao(): string {
        return this._tipoExameDescricao;
    }

    /**
     * Getter statusDescricao
     * @return {string}
     */
    public get statusDescricao(): string {
        return this._statusDescricao;
    }

    /**
     * Getter idPedidoEnriquecimento
     * @return {number}
     */
    public get idPedidoEnriquecimento(): number {
        return this._idPedidoEnriquecimento;
    }

    /**
     * Getter idPedidoContato
     * @return {number}
     */
    public get idPedidoContato(): number {
        return this._idPedidoContato;
    }

    /**
     * Setter idTarefa
     * @param {number} value
     */
    public set idTarefa(value: number) {
        this._idTarefa = value;
    }

    /**
     * Setter tipoTarefa
     * @param {number} value
     */
    public set tipoTarefa(value: number) {
        this._tipoTarefa = value;
    }

    /**
     * Setter idBusca
     * @param {number} value
     */
    public set idBusca(value: number) {
        this._idBusca = value;
    }

    /**
     * Setter rmr
     * @param {number} value
     */
    public set rmr(value: number) {
        this._rmr = value;
    }

    /**
     * Setter nomePaciente
     * @param {string} value
     */
    public set nomePaciente(value: string) {
        this._nomePaciente = value;
    }

    /**
     * Setter idSolicitacao
     * @param {number} value
     */
    public set idSolicitacao(value: number) {
        this._idSolicitacao = value;
    }

    /**
     * Setter idDoador
     * @param {number} value
     */
    public set idDoador(value: number) {
        this._idDoador = value;
    }

    /**
     * Setter tipoDoador
     * @param {number} value
     */
    public set tipoDoador(value: number) {
        this._tipoDoador = value;
    }

    /**
     * Setter dmr
     * @param {number} value
     */
    public set dmr(value: number) {
        this._dmr = value;
    }

    /**
     * Setter nomeDoador
     * @param {string} value
     */
    public set nomeDoador(value: string) {
        this._nomeDoador = value;
    }

    /**
     * Setter nomeLaboratorio
     * @param {string} value
     */
    public set nomeLaboratorio(value: string) {
        this._nomeLaboratorio = value;
    }

    /**
     * Setter idPedidoExame
     * @param {number} value
     */
    public set idPedidoExame(value: number) {
        this._idPedidoExame = value;
    }

    /**
     * Setter tipoExameDescricao
     * @param {string} value
     */
    public set tipoExameDescricao(value: string) {
        this._tipoExameDescricao = value;
    }

    /**
     * Setter statusDescricao
     * @param {string} value
     */
    public set statusDescricao(value: string) {
        this._statusDescricao = value;
    }

    /**
     * Setter idPedidoEnriquecimento
     * @param {number} value
     */
    public set idPedidoEnriquecimento(value: number) {
        this._idPedidoEnriquecimento = value;
    }

    /**
     * Setter idPedidoContato
     * @param {number} value
     */
    public set idPedidoContato(value: number) {
        this._idPedidoContato = value;
    }

    /**
     * Getter tipoSolicitacaoDescricao
     * @return {string}
     */
    public get tipoSolicitacaoDescricao(): string {
        return this._tipoSolicitacaoDescricao;
    }

    /**
     * Setter tipoSolicitacaoDescricao
     * @param {string} value
     */
    public set tipoSolicitacaoDescricao(value: string) {
        this._tipoSolicitacaoDescricao = value;
    }

    /**
     * Getter statusSolicitacaoDescricao
     * @return {string}
     */
    public get statusSolicitacaoDescricao(): string {
        return this._statusSolicitacaoDescricao;
    }

    /**
     * Setter statusSolicitacaoDescricao
     * @param {string} value
     */
    public set statusSolicitacaoDescricao(value: string) {
        this._statusSolicitacaoDescricao = value;
    }

    /**
     * Getter dataCriacao
     * @return {Date}
     */
    public get dataCriacao(): Date {
        return this._dataCriacao;
    }

    /**
     * Setter dataCriacao
     * @param {Date} value
     */
    public set dataCriacao(value: Date) {
        this._dataCriacao = value;
    }

    /**
     * Getter idPedidoContatoSms
     * @return {number}
     */
    public get idPedidoContatoSms(): number {
        return this._idPedidoContatoSms;
    }

    /**
     * Setter idPedidoContatoSms
     * @param {number} value
     */
    public set idPedidoContatoSms(value: number) {
        this._idPedidoContatoSms = value;
    }


    public jsonToEntity(res: any): this {

        this.idTarefa = ConvertUtil.parseJsonParaAtributos(res.idTarefa, new Number());
        this.tipoTarefa = ConvertUtil.parseJsonParaAtributos(res.tipoTarefa, new Number());
        this.idBusca = ConvertUtil.parseJsonParaAtributos(res.idBusca, new Number());
        this.rmr = ConvertUtil.parseJsonParaAtributos(res.rmr, new Number());
        this.nomePaciente = ConvertUtil.parseJsonParaAtributos(res.nomePaciente, new String());
        this.idSolicitacao = ConvertUtil.parseJsonParaAtributos(res.idSolicitacao, new Number());
        this.tipoSolicitacaoDescricao = ConvertUtil.parseJsonParaAtributos(res.tipoSolicitacaoDescricao, new String());
        this.statusSolicitacaoDescricao = ConvertUtil.parseJsonParaAtributos(res.statusSolicitacaoDescricao, new String());
        this.idDoador = ConvertUtil.parseJsonParaAtributos(res.idDoador, new Number());
        this.tipoDoador = ConvertUtil.parseJsonParaAtributos(res.tipoDoador, new Number());
        this.dmr = ConvertUtil.parseJsonParaAtributos(res.dmr, new Number());
        this.nomeDoador = ConvertUtil.parseJsonParaAtributos(res.nomeDoador, new String());
        this.nomeLaboratorio = ConvertUtil.parseJsonParaAtributos(res.nomeLaboratorio, new String());
        this.idPedidoExame = ConvertUtil.parseJsonParaAtributos(res.idPedidoExame, new Number());
        this.tipoExameDescricao = ConvertUtil.parseJsonParaAtributos(res.tipoExameDescricao, new String());
        this.statusDescricao = ConvertUtil.parseJsonParaAtributos(res.statusDescricao, new String());
        this.idPedidoEnriquecimento = ConvertUtil.parseJsonParaAtributos(res.idPedidoEnriquecimento, new Number());
        this.idPedidoContato = ConvertUtil.parseJsonParaAtributos(res.idPedidoContato, new Number());
        this.dataCriacao = ConvertUtil.parseJsonParaAtributos(res.dataCriacao, new Date());
        this.idPedidoContatoSms = ConvertUtil.parseJsonParaAtributos(res.idPedidoContatoSms, new Number());

        return this;
    }
}